ush/python/pygfs/task/jedi.py

```diff
@@ -0,0 +1,300 @@
+        exec_cmd.add_default_arg(self.task_config.jedi_yaml)
+
+        try:
+            logger.debug(f"Executing {exec_cmd}")
+            exec_cmd()
+        except OSError:
+            raise OSError(f"Failed to execute {exec_cmd}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
+
+        pass
+
+    @logit(logger)
+    def finalize(self) -> None:
+        super().finalize()
+
+    @logit(logger)
+    def clean(self) -> None:
+        super().clean()
+
+    @logit(logger)
+    def get_jedi_config(self, algorithm: Optional[str] = None) -> Dict[str, Any]:
+        """Compile a dictionary of JEDI configuration from template file
+
+        Parameters
+        ----------
+        algorithm (optional) : str
+            Name of the algorithm to use in the JEDI configuration. Will override the algorithm
+            set in the self.task_config.JCB_<>_YAML file
+
+        Returns
+        ----------
+        jedi_config : Dict
+            a dictionary containing the fully rendered JEDI yaml configuration
+        """
+
+        # Generate JEDI YAML file
+        logger.info(f"Generate JEDI YAML config: {self.task_config.jedi_yaml}")
+
+        if 'JCB_BASE_YAML' in self.task_config.keys():
+            # Step 1: Fill templates of the jcb base YAML file
+            jcb_config = parse_j2yaml(self.task_config.JCB_BASE_YAML, self.task_config)
+
+            # Step 2: If algorithm is present override the algorithm in the JEDI config,
+            # or, if algorithm YAML is present, fill its templates and merge.
+            if algorithm:
+                jcb_config['algorithm'] = algorithm
+            elif 'JCB_ALGO' in self.task_config.keys():
+                jcb_config['algorithm'] = self.task_config.JCB_ALGO
+            elif 'JCB_ALGO_YAML' in self.task_config.keys():
+                jcb_algo_config = parse_j2yaml(self.task_config.JCB_ALGO_YAML, self.task_config)
+                jcb_config = {**jcb_config, **jcb_algo_config}
+
+            # Step 3: Generate the JEDI Yaml using JCB driving YAML
+            self.task_config.jedi_config = render(jcb_config)
+        elif 'JEDIYAML' in self.task_config.keys():
+            # Generate JEDI YAML file (without using JCB)
+            self.task_config.jedi_config = parse_j2yaml(self.task_config.JEDIYAML, self.task_config,
+                                                        searchpath=self.gdasapp_j2tmpl_dir)
+        else:
+            raise KeyError(f"Task config must contain JCB_ALGO, JCB_BASE_YAML, or JEDIYAML")
+
+        logger.debug(f"JEDI config:\n{pformat(self.task_config.jedi_config)}")
+
+        # Save JEDI config to YAML file
+        logger.debug(f"Generate YAML file: {self.task_config.jedi_yaml}")
+        save_as_yaml(self.task_config.jedi_config, self.task_config.jedi_yaml)
+        logger.info(f"Wrote YAML to: {self.task_config.jedi_yaml}")
+
+    def link_jedi_exe(self) -> None:
```

Comment:
we probably should have this in a common utility module

Reply:
I agree